File: backend/ToDoListApp/ToDoListApp.Repository/Models/Task.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using ToDoListApp.Models.Interfaces;

namespace ToDoListApp.Repository.Models
{
    [Table("Tasks")]
    public class Task : IAuditable
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [StringLength(50, MinimumLength = 2)]
        public string Title { get; set; }

        [Required]
        [StringLength(100, MinimumLength = 2)]
        public string? Description { get; set; }

        [Required]
        [DefaultValue(false)]
        public bool IsCompleted { get; set; }
        public DateTime? CompletedOn { get; set; }
        [Required]
        public int CreatedBy { get; set; }
        public int? ModifiedBy { get; set; }
        [Required]
        public DateTime CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }

        [ForeignKey("CreatedBy")]
        public virtual User User { get; set; }
    }
}

File: frontend/to-do-list-app/src/app/models/Task.ts
export default class Task {
  id?: number;
  title: string;
  description: string;
  isCompleted: boolean;
  createdOn?: Date;
  completedOn?: Date | null;
  modifiedOn?:Date;
  constructor(
    title: string,
    description: string,
    isCompleted: boolean,
    createdOn?: Date,
    completedOn?: Date,
    id?: number
  ) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.isCompleted = isCompleted;
    this.completedOn = completedOn;
    this.createdOn = createdOn;
  }
}
